domains/certificates/Query.API/API/MeasurementsSyncer/EnergyMeasuredIntegrationEventHandler.cs

```diff
@@ -0,0 +1,112 @@
+            clearTextAttributes.Add(EnergyTagAttributeKeys.EnergyTagProductionStartingIntervalTimestamp, measurementEvent.DateFrom.ToString());
+            clearTextAttributes.Add(EnergyTagAttributeKeys.EnergyTagProductionEndingIntervalTimestamp, measurementEvent.DateTo.ToString());
+            clearTextAttributes.Add(EnergyTagAttributeKeys.EnergyTagGcFaceValue, measurementEvent.Quantity.ToString());
+            clearTextAttributes.Add(EnergyTagAttributeKeys.EnergyTagProductionDeviceUniqueIdentification, measurementEvent.GSRN);
+            clearTextAttributes.Add(EnergyTagAttributeKeys.EnergyTagConnectedGridIdentification, measurementEvent.GridArea);
+            clearTextAttributes.Add(EnergyTagAttributeKeys.EnergyTagProductionDeviceLocation, address);
+            clearTextAttributes.Add(EnergyTagAttributeKeys.EnergyTagProductionDeviceCapacity, measurementEvent.Capacity);
+            clearTextAttributes.Add(EnergyTagAttributeKeys.EnergyTagProductionDeviceCommercialOperationDate, "N/A");
+            clearTextAttributes.Add(EnergyTagAttributeKeys.EnergyTagEnergyCarrier, "Electricity");
+            clearTextAttributes.Add(EnergyTagAttributeKeys.EnergyTagGcIssueDeviceType, "Production");
+            clearTextAttributes.Add(EnergyTagAttributeKeys.EnergyTagProducedEnergySource, measurementEvent.Technology.AibFuelCode);
+            clearTextAttributes.Add(EnergyTagAttributeKeys.EnergyTagProducedEnergyTechnology, measurementEvent.Technology.AibTechCode);
+        }
+        else
+        {
+            clearTextAttributes.Add(EnergyTagAttributeKeys.AssetId, measurementEvent.GSRN);
+        }
+
+        var certificate = new CertificateDto
+        {
+            Id = Guid.NewGuid(),
+            End = measurementEvent.DateTo,
+            Start = measurementEvent.DateFrom,
+            Quantity = (uint)measurementEvent.Quantity,
+            Type = MapMeterType(measurementEvent.MeterType),
+            GridArea = measurementEvent.GridArea,
+            ClearTextAttributes = clearTextAttributes,
+            HashedAttributes = new List<HashedAttribute>()
+        };
+        return certificate;
+    }
+
+    private CertificateType MapMeterType(MeterType meterType)
+    {
+        return meterType switch
+        {
+            MeterType.Production => CertificateType.Production,
+            MeterType.Consumption => CertificateType.Consumption,
+            _ => throw new InvalidOperationException($"Invalid meter type {meterType}")
+        };
+    }
+}
+
+public class EnergyMeasuredIntegrationEventHandlerDefinition : ConsumerDefinition<EnergyMeasuredIntegrationEventHandler>
+{
+    protected override void ConfigureConsumer(
+        IReceiveEndpointConfigurator endpointConfigurator,
+        IConsumerConfigurator<EnergyMeasuredIntegrationEventHandler> consumerConfigurator,
+        IRegistrationContext context
+    )
+    {
+        endpointConfigurator.UseMessageRetry(r => r
```

Comment:
should have ADR for retry policies in general? Just so they have same'ish behaviour?

Reply:
Even better yet: regression tests in case someone does not follow our convention(s)?